presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionTreeUtils.java

```diff
@@ -147,17 +147,15 @@ public static boolean isEqualComparisonExpression(Expression expression)
         return Optional.empty();
     }
 
-    public static Optional<Object> tryResolveEnumLiteral(DereferenceExpression node, Type nodeType)
+    public static Object resolveEnumLiteral(DereferenceExpression node, Type nodeType)
     {
         QualifiedName qualifiedName = DereferenceExpression.getQualifiedName(node);
-        if (!(nodeType instanceof TypeWithName && ((TypeWithName) nodeType).getType() instanceof EnumType) || qualifiedName == null) {
```

Comment:
looks like we're no longer checking for `qualifiedName == null`, is this fine?

Reply:
The check actually hide a bug in sql function where the base was rewritten from an identifier to a variable thus qualifiedName was null. If everything works correctly then qualifiedName should never be null (it's a dereference expression after all). So if we keep the check, the check should be a `checkState`.